process/block/interceptors/headerInterceptor.go

```diff
@@ -80,10 +80,9 @@ func (hi *HeaderInterceptor) processHeader(hdrIntercepted *block.InterceptedHead
 		return
 	}
 
-	err := hi.hdrInterceptorBase.storer.Has(hdrIntercepted.Hash())
-	isHeaderInStorage := err == nil
-	if isHeaderInStorage {
-		log.Debug("intercepted block header already processed")
+	isHeaderOkForProcessing := hi.hdrInterceptorBase.headerValidator.CheckHeaderHandlerValid(hdrIntercepted.Header)
```

Comment:
hdrInterceptorBase could be made as only a reference to the base that have a simpler hi.headerValidator.CheckHeaderValid(...). Like *baseProcessor is in shardProcessor.

Reply:
hdrInterceptorBase is scheduled to be removed from the project.